src/tests/base/test_api.py

```diff
@@ -0,0 +1,78 @@
+import json
+
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from chaosinventory.authentication.models import Token, User
+
+
+class ApiTestCase(TestCase):
+    def setUp(self) -> None:
+        self.user_password: str = 'password'
+        self.user: User = get_user_model().objects.create(
+            username=self.__class__.__name__,
+            first_name='Demo',
+            last_name='User',
+            email='user@example.org',
+        )
+        self.db_token = Token()
+        self.db_token.user = self.user
+        self.db_token.application = self.__class__.__name__
+        self.db_token.save()
+        self.token = self.db_token.key
+
+    # TODO: Deduplicate
+    # Currently, this part is the same as in test_token.py
+    # but it would be nice, if we could deduplicate a bit
+    def assertValidJson(self, content: str):
+        try:
+            return json.loads(content)
+        except json.JSONDecodeError:
+            self.fail('Failed to parse JSON')
+
+    def api_call(self, path: str, payload: dict = None, token: str = None, method: str = 'get',
+                 decode_json: bool = True):
+        if payload is None:
+            payload = {}
+
+        headers = {}
+        if token is not None:
+            headers['HTTP_AUTHORIZATION'] = f'Token {token}'
+
+        if method == 'get':
+            response = self.client.get(
+                path,
+                payload,
+                **headers,
+            )
+        elif method == 'post':
+            response = self.client.post(
+                path,
+                payload,
+                **headers,
+            )
+        else:
+            raise ValueError('method must be either get or post')
+
+        if decode_json:
+            self.assertValidJson(response.content)
+            return response, json.loads(response.content)
+        else:
+            return response, response.content
+
+    # End of the copied part
```

Comment:
Do we have the possibility to deduplicate those test support methods? I've got the feeling this will lead to problems in the future.

Reply:
We could create a custom TestCase and add these methods there. Does that sound good to you?

Reply:
Alternatively, an "TestSupport"-Class/Module for stuff like API Requests and Valid JSON?

Reply:
I'm creating some mixins, python can inherit from multiple objects which allows us to have a e.g. `ApiTestMixin` which hosts the api request and json parsing code.

Reply:
I've implemented this in 40c7f92